app/src/main/java/com/example/tsaving/LoginActivity.kt

```diff
@@ -1,68 +1,76 @@
-import java.io.IOException
 import kotlin.coroutines.CoroutineContext
 
-class LoginActivity : AppCompatActivity(), CoroutineScope {
+class LoginActivity : AppCompatActivity(), CoroutineScope, LifecycleOwner {
     lateinit var job: Job
-
+    private val loginViewModel = LoginViewModel()
     override val coroutineContext: CoroutineContext get() = job + Dispatchers.Main
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_login)
+        lifecycle.addObserver(loginViewModel)
         job = Job()
 
+        //setting default event
+        et_login_email?.afterTextChanged { layout_login_email.setError(null) }
+        et_login_password?.afterTextChanged { layout_login_password.setError(null) }
         tv_login_signup.setOnClickListener {
             startActivity(Intent(this@LoginActivity, RegisterActivity::class.java))
         }
 
-        btn_login_signin.setOnClickListener {
-            var repo: TsavingRepository = TsavingRepository()
-
-            //please change email & passwordnya from edit text then delete this comment
-            var request: LoginRequestModel = LoginRequestModel("testing@apa.com", "testing")
-            Log.i("login req :", request.toString())
+        loginViewModel.apply {
+            flagStatus.observe(this@LoginActivity, Observer {
+                if(flagStatus.value == ErrorName.NullEmailAndPass){
+                    layout_login_email.setError("Please Input Email Field")
+                    layout_login_password.setError("Please Input Password Field")
+                }
+                else if(flagStatus.value == ErrorName.NullEmail){
+                    layout_login_email.setError("Please Input Email Field")
+                }
+                else if(flagStatus.value == ErrorName.NullPassword){
+                    layout_login_password.setError("Please Input Password Field")
+                }
+                else if(flagStatus.value == ErrorName.InvalidEmail){
+                    layout_login_email.setError("Invalid Email Format")
+                }
+                else if(flagStatus.value == ErrorName.ErrorNetwork){
+                    DialogHandling().basicAlert(this@LoginActivity, "Notification", "Network Error", "close")
+                }
+                else if(flagStatus.value == ErrorName.InvalidLogin){
+                    DialogHandling().basicAlert(this@LoginActivity, "Notification", "Wrong Username Or Password", "close")
+                }
+            })
+            dataLogin.observe(this@LoginActivity, Observer {
+                if(it.status == "SUCCESS"){
+                    //save token
+                    BaseApplication.token = it.data.token
+                    finish()
+                    startActivity(Intent(this@LoginActivity, MainActivity::class.java))
+                    Toast.makeText(applicationContext, "Login Success", Toast.LENGTH_SHORT).show()
```

Comment:
i dont think we need this message because theres already enough feedback when changing screen to dashboard

Reply:
ok